lib/bitcoin/BitcoinRawDataParser.ts

```diff
@@ -0,0 +1,85 @@
+import { Block } from 'bitcore-lib';
+
+/**
+ * Parser for raw bitcoin block data
+ */
+export default class BitcoinRawDataParser {
+
+  /**
+   * The beginning of each block contains the magic bytes indicating main or test net
+   * followed by a 4 byte number indicating how big the block data is
+   */
+  private static magicBytes = {
+    testnet: '0b110907',
+    mainnet: 'f9beb4d9'
+  };
+  private static sizeBytesLength = 4;
+
+  /**
+   * Parse the given raw block data file. It can throw error if block data is invalid when validating magic bytes,
+   * creating new Block, or validating size
+   * @param rawBlockDataFileBuffer The file, in buffer form, to be parsed as blocks
+   */
+  public static parseRawDataFile (rawBlockDataFileBuffer: Buffer): any {
+    let rawBlockDataFileString = rawBlockDataFileBuffer.toString('hex');
+    let hexStrings: string[];
+    let magicBytes: string;
+    const blockMapper: any = {};
+
+    // split the hex by magic bytes
+    if (rawBlockDataFileString.startsWith(BitcoinRawDataParser.magicBytes.testnet)) {
+      magicBytes = BitcoinRawDataParser.magicBytes.testnet;
+      hexStrings = rawBlockDataFileString.split(magicBytes);
+    } else if (rawBlockDataFileString.startsWith(BitcoinRawDataParser.magicBytes.mainnet)) {
+      magicBytes = BitcoinRawDataParser.magicBytes.mainnet;
+      hexStrings = rawBlockDataFileString.split(magicBytes);
+    } else {
+      throw new Error('Invalid block data');
+    }
+
+    // remove the first empty string after split
+    hexStrings.shift();
+
+    let currentHex = '';
+    let count = 0;
+    for (const hexString of hexStrings) {
+      if (currentHex === '') {
+        currentHex = hexString;
+      } else {
+        // take care of when non magic bytes get split. Add back the bytes and concat
+        currentHex = `${currentHex}${magicBytes}${hexString}`;
+      }
+      if (BitcoinRawDataParser.verifySize(currentHex)) {
+        // A hex string can be treated as raw block data once size is verified and size bytes removed
+        const rawBlockDataAsString = currentHex.slice(BitcoinRawDataParser.sizeBytesLength * 2);
+        const rawBlockDataAsBuffer = Buffer.from(rawBlockDataAsString, 'hex');
```

Comment:
Design consideration: is it possible work on Buffer as opposed to HEX string? That would avoid code that converts to HEX, then back to buffer again, the * 2 etc.

Reply:
Good point. Will switch to operating with buffers. I used string thinking would be easier to read.